src/services/api/articles.js

```diff
@@ -1,45 +1,59 @@
 
-import ApiBase, { tryCastToApiError } from './api-base';
+import ApiBase, { tryCastToApiError, toUrlParam, toUrlParams, isValid} from './api-base';
 
 export default class ArticlesApi extends ApiBase
 {
   constructor(options) {
     super(options);
   }
   
-  async queryArticleGroup(groupKey, params)  {
+  async queryArticleGroup(groupKey, {q, f, s, sk, l , c, fo, ag })  {    
+    if(!isValid(groupKey)) throw Error(`invalid value for groupKey`); 
+    const params = toUrlParams( {q, f, s, sk, l , c, fo, ag });
+
     return this.http.get(`api/v2017/articles/grouping/${encodeURIComponent(groupKey)}`, { params })
-                    .then(res => res.data)
-                    .catch(tryCastToApiError);
-  }
+                     .then(res => res.data)
+                     .catch(tryCastToApiError);
+   }
+
+  async queryArticles({q, f, s, sk, l , c, fo, ag })  {
+    const params = toUrlParams( {q, f, s, sk, l , c, fo, ag });
 
-  async queryArticles(params)  {
     return this.http.get(`api/v2017/articles`, { params })
                     .then(res => res.data)
                     .catch(tryCastToApiError);
   }
 
   async getArticleById(id)  {
+      if(!isValid(id)) throw Error(`invalid value for id`);
+      const params= {
+        q: toUrlParam( { _id: mapObjectId(id) })
```

Comment:
ideally, id  will be a string value, so no need to call toUrlParam

Reply:
There is an error in code. 
id parameter should pass in url address, so after editing code, no more this issue.